fe/fe-core/src/main/java/com/starrocks/sql/optimizer/validate/MVChecker.java

```diff
@@ -0,0 +1,96 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.optimizer.validate;
+
+import com.google.api.client.util.Lists;
+import com.google.common.base.Joiner;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.sql.PlannerProfile;
+import com.starrocks.sql.optimizer.OptExpression;
+import com.starrocks.sql.optimizer.operator.Operator;
+import com.starrocks.sql.optimizer.operator.physical.PhysicalScanOperator;
+import com.starrocks.sql.optimizer.rule.transformation.materialization.MaterializedViewRewriter;
+import com.starrocks.sql.optimizer.task.TaskContext;
+
+import java.util.List;
+import java.util.Map;
+
+public class MVChecker implements PlanValidator.Checker {
```

Comment:
It seems not used to check whether the plan is valid. Maybe consider another name?

Reply:
This will be used later when we support `mv_rewrite_mode`: https://github.com/StarRocks/starrocks/pull/26997

Do you have a better name?

Reply:
How about directly using `MVRewriteInfoCollector`